allure-junit5/src/test/java/io/qameta/allure/examples/junit5/AllureStepTest.java

```diff
@@ -0,0 +1,35 @@
+package io.qameta.allure.examples.junit5;
+
+import io.qameta.allure.Allure;
+import io.qameta.allure.Step;
+import org.junit.jupiter.api.Test;
+
+public class AllureStepTest {
+
+    private static final String GLOBAL_PARAMETER = "global value";
+
+    @Test
+    public void annotatedStepTest() {
+        annotatedStep("local value");
+    }
+
+    @Test
+    public void lambdaStepTest() {
+        final String localParameter = "parameter value";
+        Allure.step(String.format("Parent lambda step with parameter [%s]", localParameter), (step) -> {
+            step.parameter("parameter", localParameter);
+            Allure.step(String.format("Nested lambda step with global parameter [%s]", GLOBAL_PARAMETER));
+        });
+    }
+
+    @Step("Parent annotated step with parameter [{parameter}]")
+    public void annotatedStep(final String parameter) {
+        nestedAnnotatedStep();
+    }
```

Comment:
using @Step annotation without parameter is also a valuable example to cover, especially taking into account that depending on language (java, python, c#) the behaviour is different... 

```
    @Step
    public void annotatedStepWithAutomaticStepName(final String parameter) {
        // ...
    }
```

Reply:
depending on whether allure java commons support automatic reflection of step params in case @Step annontation does not receive an argument – we might be interested to reflect this case also...